### DISCUSSION TITLE
Strange output in obfuscated JAR

### DISCUSSION THREAD

**Discussion by khleebi** [CastleLab/COMP3021-2021Fall-PA1-Student-Version#32] [Q&A] [answered]
I have created a game file with the following settings:

3
3
1
WWW.
WWW.
WGP.
....

Then I run the game with the newest obfuscated JAR I get the following output:
![output](https://user-images.githubusercontent.com/90165647/135475523-bff1b5ba-badb-4c75-8f33-e9acf16db6a4.png)

I expect there should be an exception thrown before the game starts because those empty cells disappear, but it runs as usual.

Not sure whether it is a reasonable output or a bug. Please let me know if I misunderstand anything.

**Comment by Derppening** [comment, marked as answer]
This is expected behavior. When implementing the `GameSerializer` class, I ignore all characters which are outside the boundaries of the gameboard size declared in the first two lines.

In other words, since this part is not implemented by you, you don't need to care about it.
